### ISSUE TITLE
Document limitations of Servlet Filter observations

### ISSUE THREAD

**Issue by jonatan-ivanov** [spring-projects/spring-framework#29398]
**Affects:** 6.0.0-SNAPSHOT

If micrometer-observation is set-up and if I throw an exception from a controller:
```java
@GetMapping("/trouble")
String trouble() {
    LOGGER.info("3,2,1... Boom!");
    throw new IllegalArgumentException("Noooooo!");
}
```

The `ERROR` log entry does not contain the traceID/spanId, see `[boot3-sample,,]` (log correlation does not work):
```
2022-10-28T12:37:16.647-07:00  INFO [boot3-sample,635c2f6c8cb895fa15f922f7cacc6954,15f922f7cacc6954] 96088 --- [nio-8080-exec-1] i.m.boot3.samples.SampleController       : 3,2,1... Boom!
2022-10-28T12:37:16.660-07:00 ERROR [boot3-sample,,] 96088 --- [nio-8080-exec-1] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: java.lang.IllegalArgumentException: Noooooo!] with root cause

java.lang.IllegalArgumentException: Noooooo!
...
```

I think the issue is either the error logging is not "observed" or it is not "in-Scope".

**Comment by bclozel** [comment]
Hey @jonatan-ivanov 

From what I understand, this is being logged from an `org.apache.catalina` class? In this case, it's called outside of the observation scope of the Servlet filter so it's expected at this stage not to have the log correlation anymore.

I don't think Framework can do anything about that.

**Comment by jonatan-ivanov** [comment]
Yes, it is definitely out of the scope of the Observation.
I'm not sure if it is purely Catalina or a component that we provide to Catalina (`dispatcherServlet` in the name of the logger).

@marcingrzejszczak might have more details, in Sleuth, we fixed it with a `Valve`: [TraceValve](https://github.com/spring-cloud/spring-cloud-sleuth/blob/3.1.x/spring-cloud-sleuth-instrumentation/src/main/java/org/springframework/cloud/sleuth/instrument/web/tomcat/TraceValve.java)

**Comment by jonatan-ivanov** [comment]
I think having traceId/spanId on error logs is a vital feature.
Errors are the events where users most likely want to look into, with missing traceId/spanId they are unable to do so out of the box.

**Comment by jonatan-ivanov** [comment]
I only used the valve as an example (Tomcat), I'm not saying we should use it or that would be a universal solution. I'm only saying that out of the box this does not work and this is a rather important functionality.

I guess the question for me is: can we do anything with this in the 6.0 timeline or in 6.1? At least discuss it with the whole team?

If this can't be fixed in 6.0, can we document it? I guess other options should work too, like having a ControllerAdvice or using a custom [ObservationHandler](https://github.com/micrometer-metrics/micrometer-samples/blob/d700c5b70d398373c19e3deaf7510b2bc6a5b69e/micrometer-samples-boot3-database/src/main/java/io/micrometer/boot3/samples/db/Boot3WithDatabaseSampleApplication.java#L33-L48).

**Comment by bclozel** [comment]
I think we need to document this as a known behavior in our web observability support documentation. Since the instrumentation is done at the Servlet filter level, we can't extend the observation outside of this scope. Applications and libraries can write container-specific instrumentation with broader support but this won't be supported at the Spring Framework level.
